cstore_fdw.c

```diff
@@ -532,6 +531,45 @@ DroppedCStoreFilenameList(DropStmt *dropStatement)
 }
 
 
+/* FindCStoreTables returns list of CStore tables from given table list */
+static List *
+FindCStoreTables(List *tableList)
+{
+	List *cstoreTableList = NIL;
+	ListCell *relationCell = NULL;
+	foreach(relationCell, tableList)
+	{
+		RangeVar *rangeVar = (RangeVar *) lfirst(relationCell);
+
+		Oid relationId = RangeVarGetRelid(rangeVar, AccessShareLock, true);
+		if (CStoreTable(relationId))
+		{
+			cstoreTableList = lappend(cstoreTableList, rangeVar);
+		}
+	}
+
+	return cstoreTableList;
+}
+
+/* TruncateCStoreTable truncates given cstore tables */
+static void
+TruncateCStoreTables(List *cstoreTableList)
+{
+	ListCell *relationCell = NULL;
+	foreach(relationCell, cstoreTableList)
+	{
+		RangeVar *rangeVar = (RangeVar *) lfirst(relationCell);
```

Comment:
btw, I was able to crash backend with concurrent truncate & copy commands with the lack of `AccessExclusiveLock`.